subgraphs/burrow/src/handlers/actions.ts

```diff
@@ -0,0 +1,589 @@
+  hourlySnapshot.save();
+  market.save();
+  account.save();
+  deposit.save();
+  position.save();
+  protocol.save();
+
+  updateProtocol();
+}
+
+/**
+ * @dev Handles deposits made to reserve pool
+ * @notice Admin/Owner can deposit to reserve
+ * @param event args {amount, account_id, token_id}
+ */
+export function handleDepositToReserve(event: EventData): void {
+  const parsedData = parse0(event.data);
+  const amount = parsedData[1];
+  const token_id = parsedData[2];
+
+  const market = getOrCreateMarket(token_id);
+
+  market._totalReserved = market._totalReserved.plus(
+    BigDecimal.fromString(amount)
+  );
+
+  // we keep the reserve added to the pool seperately for revenue calculation
+  market._added_to_reserve = market._added_to_reserve.plus(
+    BigDecimal.fromString(amount)
+  );
+
+  market.save();
+}
+
+/**
+ * Handles withdraws from pool
+ * @param event args {account_id, amount, token_id}
+ */
+export function handleWithdraw(event: EventData): void {
+  const receipt = event.receipt;
+  const logIndex = event.logIndex;
+  const data = event.data;
+
+  const withdraw = getOrCreateWithdrawal(
+    receipt.outcome.id
+      .toBase58()
+      .concat("-")
+      .concat((logIndex as i32).toString()),
+    receipt
+  );
+  withdraw.logIndex = logIndex as i32;
+  const parsedData = parse0(data);
+  const account_id = parsedData[0];
+  const amount = parsedData[1];
+  const token_id = parsedData[2];
+
+  const market = getOrCreateMarket(token_id.toString());
+  const dailySnapshot = getOrCreateMarketDailySnapshot(market, receipt);
+  const hourlySnapshot = getOrCreateMarketHourlySnapshot(market, receipt);
+  const usageDailySnapshot = getOrCreateUsageMetricsDailySnapshot(receipt);
+  const usageHourlySnapshot = getOrCreateUsageMetricsHourlySnapshot(receipt);
+  const financialDailySnapshot = getOrCreateFinancialDailySnapshot(receipt);
+  const account = getOrCreateAccount(account_id.toString());
+  const position = getOrCreatePosition(account, market, "LENDER", receipt);
+
+  const token = getOrCreateToken(token_id.toString());
+  withdraw.account = account.id;
+  withdraw.amount = BigInt.fromString(amount.toString());
+  withdraw.asset = token.id;
+  withdraw.market = market.id;
+  withdraw.position = position.id;
+
+  withdraw.amountUSD = withdraw.amount
+    .divDecimal(
+      BigInt.fromI32(10)
+        .pow((token.decimals + token.extraDecimals) as u8)
+        .toBigDecimal()
+    )
+    .times(token.lastPriceUSD!);
+
+  // usage metrics
+  if (account._last_active_timestamp.lt(usageDailySnapshot.timestamp)) {
+    usageDailySnapshot.dailyActiveUsers += 1;
+    usageHourlySnapshot.hourlyActiveUsers += 1;
+  }
+  usageDailySnapshot.dailyWithdrawCount += 1;
+  usageDailySnapshot.dailyTransactionCount += 1;
+  usageHourlySnapshot.hourlyWithdrawCount += 1;
+  usageHourlySnapshot.hourlyTransactionCount += 1;
+
+  position.withdrawCount += 1;
+
+  // close if balance is zero
+  if (position.balance.lt(withdraw.amount)) {
```

Comment:
should be <= otherwise you are potentially missing closed positions

Reply:
will do